File: Easy/1165 - Single-Row Keyboard/1165Soln.js
'use strict'

/**
 * @param {string} keyboard
 * @param {string} word
 * @return {number}
 */
const calculateTime = function(keyboard, word) {
    let sum = 0
    let prevIdx = 0
    let tmpIdx
    for (let i = 0; i < word.length; i++) {
        tmpIdx = keyboard.indexOf(word[i])
        sum += Math.abs(tmpIdx - prevIdx)
        prevIdx = tmpIdx
    }

    return sum
}

File: Easy/1165 - Single-Row Keyboard/1165Soln.rb
# frozen_string_literal: true

# @param {String} keyboard
# @param {String} word
# @return {Integer}
def calculate_time(keyboard, word)
  prev_idx = 0
  sum = 0
  word.split('').each do |c|
    tmp_idx = keyboard.index(c)
    sum += (prev_idx - tmp_idx).abs
    prev_idx = tmp_idx
  end
  sum
end
